WordPressAuthenticator/Analytics/SignInTracker.swift

```diff
@@ -0,0 +1,345 @@
+import Foundation
+
+/// Implements the analytics tracking logic for our sign in flow.
+///
+public class SignInTracker {
+    
+    /// The method used for analytics tracking.  Useful for overriding in automated tests.
+    ///
+    typealias TrackerMethod = (_ event: AnalyticsEvent) -> ()
+
+    enum EventType: String {
+        case step = "unified_login_step"
+        case interaction = "unified_login_interaction"
+        case failure = "unified_login_failure"
+    }
+    
+    enum Property: String {
+        case failure
+        case flow
+        case click
+        case source
+        case step
+    }
+    
+    enum Source: String {
+        case `default`
+        case jetpack
+        case share
+        case deeplink
+        case reauthentication
+        
+        /// Starts when the used adds a site from the site picker
+        ///
+        case selfHosted
+    }
+    
+    enum Flow: String {
+        /// The initial flow before we decide whether the user is logging in or signing up
+        case wpCom = "wordpress_com"
+        
+        /// The flow that starts when the user starts the Google login
+        ///
+        case googleLogin = "google_login"
+        
+        case googleSignup = "google_signup"
+        
+        /// Flow for Sign in with Apple
+        ///
+        case apple = "siwa_login"
+        
+        /// Sign in with the icloud keychain
+        ///
+        case keychain = "icloud_keychain_login"
```

Comment:
AFAIK, icloud_keychain can't be used during sign up, so this event by itself is correct.

@ScoutHarris to confirm / deny that ☝️  is true

Reply:
Btw, I'm good with changing the name of it if we feel `iCloudLogin` or something similar would be more consistent.

Reply:
@mindgraffiti is again correct here - keychain is only used for login.

I do like making this one a bit more specific and consistent. To differentiate from the _other_ keychain, maybe we can go with something like `iCloudKeychainLogin`?

Reply:
The flow is now named `iCloudKeychainLogin`.